src/components/user-defined-sets/sets.js

```diff
@@ -0,0 +1,159 @@
+import { Set as ImmutableSet, OrderedMap as ImmutableOrderedMap } from 'immutable';
+
+// Class for storing Set objects of string IDs (cell IDs, gene IDs, etc...).
+// The collection of sets is stored as a map whose values are Set objects.
+// Think of the static methods here as reducer action functions.
+// Motivation for using reducer pattern:
+// https://medium.com/@dan_abramov/you-might-not-need-redux-be46360cf367
+
+export default class Sets {
+    static initialState = {
+      namedSets: ImmutableOrderedMap(),
+      currentSet: ImmutableSet(),
+    };
+
+    /**
+     * Modifies the set entry for a named set.
+     * @param {object} state The current state.
+     * @param {string} key The set key.
+     * @param {Set} set The set value.
+     * @returns {object} The new state.
+     */
+    static setNamedSet(state, key, set) {
```

Comment:
Maybe `name` instead of `key`? More suggestive of semantics, and less about implementation?